File: frontend/types/leaderboard.ts
export type LeaderboardEntry = {
  rank: number;
  athlete_id: string;
  assessment_id: string;
  upload_id: string;
  name: string;
  sport: string;
  state: string;
  overall_score: number;
  technique_score: number;
  efficiency_score: number;
  balance_score: number;
  completed_at: string;
  is_current_user: boolean;
};

export type LeaderboardResponse = {
  top_athletes: LeaderboardEntry[];
  current_user_entry: LeaderboardEntry | null;
};

export type LeaderboardFilters = {
  sport?: string;
  state?: string;
  gender?: string;
  min_age?: number;
  max_age?: number;
  min_experience?: number;
  max_experience?: number;
};

File: backend/app/schemas/leaderboard.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class LeaderboardEntry(BaseModel):
    rank: int = Field(ge=1)
    athlete_id: uuid.UUID
    assessment_id: uuid.UUID
    upload_id: uuid.UUID
    name: str
    sport: str
    state: str
    overall_score: float = Field(ge=0, le=100)
    technique_score: float = Field(ge=0, le=100)
    efficiency_score: float = Field(ge=0, le=100)
    balance_score: float = Field(ge=0, le=100)
    completed_at: datetime
    is_current_user: bool


class LeaderboardResponse(BaseModel):
    top_athletes: list[LeaderboardEntry]
    current_user_entry: LeaderboardEntry | None = None
